cluster/aws/config-default.sh

```diff
@@ -15,10 +15,32 @@
 # limitations under the License.
 
 ZONE=${KUBE_AWS_ZONE:-us-west-2a}
-MASTER_SIZE=${MASTER_SIZE:-t2.micro}
-MINION_SIZE=${MINION_SIZE:-t2.micro}
+MASTER_SIZE=${MASTER_SIZE:-}
+MINION_SIZE=${MINION_SIZE:-}
 NUM_MINIONS=${NUM_MINIONS:-4}
 
+# Dynamically set node sizes so that Heapster has enough space to run
+if [[ -z ${MINION_SIZE} ]]; then
+  if (( ${NUM_MINIONS} < 50 )); then
+    MINION_SIZE="t2.micro"
+  elif (( ${NUM_MINIONS} < 150 )); then
+    MINION_SIZE="t2.small"
+  else
+    MINION_SIZE="t2.medium"
+  fi
+fi
+
+# Dynamically set the master size by the number of nodes
```

Comment:
I'm fine with this -- in fact I think it's a good idea -- but note that unlike the minion size, this isn't really based on any data AFAIK. That is, we haven't characterized what master size we need for different cluster sizes.